src/java.base/share/classes/java/lang/doc-files/ValueBased.html

```diff
@@ -31,36 +31,40 @@
-        of any other object or variable;</li>
-    <li>make no use of identity-sensitive operations such as reference
-        equality (<code>==</code>) between instances, identity hash code of
-        instances, or synchronization on an instances's intrinsic lock;</li>
-    <li>are considered equal solely based on <code>equals()</code>, not
-        based on reference equality (<code>==</code>);</li>
-    <li>do not have accessible constructors, but are instead instantiated
-        through factory methods which make no commitment as to the identity
-        of returned instances;</li>
-    <li>are <em>freely substitutable</em> when equal, meaning that interchanging
-        any two instances <code>x</code> and <code>y</code> that are equal
-        according to <code>equals()</code> in any computation or method
-        invocation should produce no visible change in behavior.
-    </li>
+        solely from the values of the class's instance fields (and properties of
+        the objects they reference), not from the instance's identity</li>
+    <li>have instances that are <em>freely substitutable</em> when equal, meaning
+        that interchanging any two instances <code>x</code> and <code>y</code> that
+        are equal according to <code>equals()</code> produces no visible change in
+        the behavior of the class's methods;</li>
+    <li>perform no synchronization using an instance's monitor;</li>
+    <li>do not have (or have deprecated any) accessible constructors;</li>
+    <li>do not provide any other instance creation mechanism that promises
+        a unique identity on each method call&mdash;in particular, any factory
+        method must allow for the possibility that if two independently-produced
+        instances are equal according to <code>equals()</code>, they may also be
+        equal according to <code>==</code>;</li>
+    <li>are final;</li>
+    <li>extend either <code>Object</code> or a hierarchy of abstract classes that
+        declare no instance fields or instance initializers and whose constructors
+        are empty.</li>
+    <li>
 </ul>
 
-<p>A program may produce unpredictable results if it attempts to distinguish two
-    references to equal values of a value-based class, whether directly via reference
+<p>When two instances of a value-based class are `equal`, a program should not attempt
+    to distinguish between their identities, whether directly via reference
     equality or indirectly via an appeal to synchronization, identity hashing,
-    serialization, or any other identity-sensitive mechanism.  Use of such
-    identity-sensitive operations on instances of value-based classes may have
-    unpredictable effects and should be avoided.</p>
+    serialization, or any other identity-sensitive mechanism.</p>
+<p>Synchronization on instances of value-based classes is strongly discouraged,
+    because the programmer cannot guarantee exclusive ownership of the
+    associated monitor.</p>
```

Comment:
I'll revise to clarify that the bulleted list is about properties of the class declaration, while the subsequent sentences are constraints on/advice to clients. (Is there a better way to talk about clients than "a program should not..."? "Programmer", "client", "user" all kind of work, but I don't love any of them...) With that distinction in mind, I don't think the client stuff works very well until we first define what we mean by "value-based class".

I did remove what seemed to me like a redundant bullet from the original list (see my earlier comment). With what's left, each one is trying to say something distinct, although maybe some rephrasing in certain places could help.

The `final` class and field restrictions were combined before, but I split them because they're two very different constraints—one about extension, the other about immutability. (It's not ideal that Java uses the same keyword for both properties.) Actually, though, it works pretty well to combine the final class restriction with the superclass restriction, leaving one bullet all about subclassing. I'll do that.

The bullets that talk about equality are saying 1) the class needs an appropriate equals/hashCode/toString; 2) the class's instance methods don't have distinct behaviors for instances that are `equals`; 3) the class's factories do not promise distinct identities for instances that are `equals`. Those are three distinct things to say; I'm not seeing a clear way to combine them.

Synchronization: forget primitive classes. This is advice to current clients of value-based classes. And the advice is: don't synchronize, because you can't be sure someone else isn't doing the same on the same object. This is advice that is relevant to current clients without asking them to imagine a future in which the Object & monitor model has changed. Yet the implication is the same: don't do it.